### PR TITLE
API-32864 Validate token ICN in Higher-Level Reviews API GET endpoints

### PR DESCRIPTION
**Pull request by caseywilliams**
## Summary

Updates the Higher-Level Reviews API v0 as follows:
1. When using a veteran-scoped token, requests will now get a 403 error when attempting to access the `show` or `download` endpoints for a record that doesn't belong to them, even if they can provide the correct `icn` parameter at the `download` endpoint, for example.
2. The `icn` parameter (which is required at the `download` endpoint to ensure that the client has the PII associated with the record before allowing them to download the PDF) is now optional when the request is using a veteran-scoped token to request a PDF of the veteran's own HLR. The parameter is still required when using a system- or representative-scoped token.

## Related issue(s)
- [API-32864](https://jira.devops.va.gov/browse/API-32864)

## Testing done
- Updated docs specs extensively
- Added a few new regular specs

## What areas of the site does it impact?
Higher-Level Reviews API v0 (not released in production yet)

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

File: modules/appeals_api/app/swagger/higher_level_reviews/v0/swagger.json [hunk 1]
```diff
@@ -793,17 +813,37 @@
...
+                    }
+                  ]
+                },
+                "schema": {
+                  "$ref": "#/components/schemas/errorModel"
+                }
+              }
+            }
+          },
+          "403": {
```
Comment (caseywilliams) [inline modules/appeals_api/app/swagger/higher_level_reviews/v0/swagger.json:836]: Initially, I started to write several more docs specs for other scopes in these APIs - examples with system tokens, representatives, veterans, etc. - but I realized that the UI does not actually render curl requests for each example - it only does that once for each endpoint, and all of the example responses just show the body that's returned - not the scopes or parameters that were sent. So I kept the number of example errors smaller and tried to communicate about scopes and params using the `description` fields instead.

File: modules/appeals_api/app/swagger/appealable_issues/v0/swagger.json [hunk 2]
```diff
@@ -365,7 +365,7 @@
                       "errors": [
                         {
                           "title": "Missing parameter",
-                          "detail": "The required parameter \"icn\", is missing",
```
[the remark this answers is not in the document]
Comment (kristen-brown) [reply modules/appeals_api/app/swagger/appealable_issues/v0/swagger.json:368]: Good call.

**Review by kristen-brown — APPROVED** [review]
Nice work! 🎉

**Review by michelpmcdonald — APPROVED** [review]
nice

**Pull request merged**